Fails to build - error: 'RegexMatcher' does not name a type

This once worked fine for me, but trying to rebuild recently and it now fails. Maybe due to now using later software packages. For example gcc 7.3.1. I'm running Arch Linux. Let me know what other information you might need. Thanks.

```
Making all in src
make[3]: Entering directory '/build/naemon/src/naemon-1.0.6/naemon-livestatus/src'
  CXX      livestatus_la-AndingFilter.lo
  CXX      livestatus_la-Column.lo
  CXX      livestatus_la-ColumnsColumn.lo
  CXX      livestatus_la-CustomVarsExplicitColumn.lo
  CXX      livestatus_la-ContactsColumn.lo
  CXX      livestatus_la-CustomVarsColumn.lo
In file included from CustomVarsColumn.cc:28:0:
CustomVarsFilter.h:45:5: error: 'RegexMatcher' does not name a type; did you mean 'uregex_matches'?
     RegexMatcher * _regex_matcher;
     ^~~~~~~~~~~~
     uregex_matches
make[3]: *** [Makefile:798: livestatus_la-CustomVarsColumn.lo] Error 1
make[3]: Leaving directory '/build/naemon/src/naemon-1.0.6/naemon-livestatus/src'
make[2]: *** [Makefile:475: all-recursive] Error 1
make[2]: Leaving directory '/build/naemon/src/naemon-1.0.6/naemon-livestatus'
make[1]: *** [Makefile:385: all] Error 2
make[1]: Leaving directory '/build/naemon/src/naemon-1.0.6/naemon-livestatus'
make: *** [Makefile:24: naemon-livestatus] Error 2
```

--- a/src/CustomVarsFilter.cc
+++ b/src/CustomVarsFilter.cc
@@ -68,8 +68,8 @@ with spaces
             }
             else {
 #ifdef HAVE_ICU
-                UnicodeString s = UnicodeString::fromUTF8(search_space);
-                _regex_matcher = new RegexMatcher(s, (_opid == OP_REGEX_ICASE ? UREGEX_CASE_INSENSITIVE : 0), status);
+                icu::UnicodeString s = icu::UnicodeString::fromUTF8(search_space);
+                _regex_matcher = new icu::RegexMatcher(s, (_opid == OP_REGEX_ICASE ? UREGEX_CASE_INSENSITIVE : 0), status);
                 if (U_FAILURE(status))
                 {
                     setError(RESPONSE_CODE_INVALID_HEADER, "invalid regular expression '%s'", value);
@@ -122,7 +122,7 @@ bool CustomVarsFilter::accepts(void *data)
             case OP_REGEX_ICASE:
 #ifdef HAVE_ICU
                 if ( _regex_matcher != 0) {
-                    UnicodeString s = UnicodeString::fromUTF8(act_string);
+                    icu::UnicodeString s = icu::UnicodeString::fromUTF8(act_string);
                     _regex_matcher->reset(s);
                     _regex_matcher->reset(act_string);
                     pass = _regex_matcher->find();

--- a/src/CustomVarsFilter.h
+++ b/src/CustomVarsFilter.h
@@ -42,7 +42,7 @@ class CustomVarsFilter : public Filter
     bool _negate;
     string _ref_text;
 #ifdef HAVE_ICU
-    RegexMatcher * _regex_matcher;
+    icu::RegexMatcher * _regex_matcher;
 #else
     regex_t *_regex;
 #endif

--- a/src/StringColumnFilter.cc
+++ b/src/StringColumnFilter.cc
@@ -50,8 +50,8 @@
         }
         else {
 #ifdef HAVE_ICU
-            UnicodeString s = UnicodeString::fromUTF8(value);
-            _regex_matcher = new RegexMatcher(s, (_opid == OP_REGEX_ICASE ? UREGEX_CASE_INSENSITIVE : 0), status);
+            icu::UnicodeString s = icu::UnicodeString::fromUTF8(value);
+            _regex_matcher = new icu::RegexMatcher(s, (_opid == OP_REGEX_ICASE ? UREGEX_CASE_INSENSITIVE : 0), status);
             if (U_FAILURE(status)) {
                 setError(RESPONSE_CODE_INVALID_HEADER, "invalid regular expression '%s'", value);
                 delete _regex_matcher;
@@ -98,7 +98,7 @@ bool StringColumnFilter::accepts(void *data)
         case OP_REGEX_ICASE:
 #ifdef HAVE_ICU
             if ( _regex_matcher != 0 ) {
-                UnicodeString s = UnicodeString::fromUTF8(act_string);
+                icu::UnicodeString s = icu::UnicodeString::fromUTF8(act_string);
                 _regex_matcher->reset(s);
                 pass = _regex_matcher->find();
             }

--- a/src/StringColumnFilter.h
+++ b/src/StringColumnFilter.h
@@ -47,7 +47,7 @@ class StringColumnFilter : public Filter
     int _opid;
     bool _negate;
 #ifdef HAVE_ICU
-    RegexMatcher *_regex_matcher;
+    icu::RegexMatcher *_regex_matcher;
 #else
     regex_t *_regex;
 #endif

